common/display/displayqueue.cpp

```diff
@@ -149,27 +149,35 @@ bool DisplayQueue::ApplyPendingModeset(drmModeAtomicReqPtr property_set) {
   return true;
 }
 
-bool DisplayQueue::SetDpmsMode(uint32_t dpms_mode) {
+bool DisplayQueue::SetPowerMode(uint32_t power_mode) {
   ScopedSpinLock lock(spin_lock_);
-  if (dpms_mode_ == dpms_mode)
+
+  if (dpms_mode_ == power_mode)
     return true;
 
-  dpms_mode_ = dpms_mode;
-  if (dpms_mode_ == DRM_MODE_DPMS_OFF) {
+  if ((power_mode == kOff) || (power_mode == kDoze)) {
+    dpms_mode_ = DRM_MODE_DPMS_OFF;
+    Flush();
```

Comment:
Ideally the size of the queue should be zero. The off/doze/doze suspend will get called only when there is no activity in the current app/the device is in idle mode. Flush is just the ideal way of handling the queue exit just added to cover any corner case for some reason the flip delayed.